Create docker image with Mantid static analysis tools

DavidFair opened the pull request:
Creates a docker image which only contains the static analysis tools
required by Mantid.
This decouples our static analysis pipeline from the builders and deps, so we can update each seperately and deploy them to different machines.

DanNixon commented on static_analysis/Dockerfile:
```diff
@@ -0,0 +1,86 @@
+# Version 1.90
+ARG cppcheck_commit=077e652
+
+RUN \
```
Doesn't this (empty line after `RUN`) generate a warning when building? It definitely used to.

DavidFair replied:
No warning, but it's wrong and a hangover from a restructuring I did mid-way

DanNixon commented on static_analysis/Dockerfile:
```diff
@@ -0,0 +1,86 @@
+
+# Build
+WORKDIR llvm/build
+RUN cmake -GNinja -DLLVM_BUILD_STATIC=ON -DLLVM_ENABLE_LIBCXX=ON ..
```
Given that `jenkins/jnlp-slave` is Debian based is there a reason we are not just using the pre-built binaries for Kitware?

DavidFair replied:
Debian stretch's version of `clang-format` is too old, I can't seem to find Kitware's PPA
Could you post a link to it as I'd rather drop this if possible

DavidFair commented:
Note to self: need to add doxygen

DavidFair commented:
This should be ready for review again

DavidFair commented:
Need to add the latest version of CMake yet....

DanNixon commented on static_analysis/Dockerfile:
```diff
@@ -0,0 +1,42 @@
+RUN apt-get install clang-format-6.0 -y --no-install-recommends
+RUN apt-get install cmake -y --no-install-recommends
+RUN apt-get install doxygen -y --no-install-recommends
+RUN apt-get install python3 python3-pip python3-setuptools -y --no-install-recommends
```
What is setuptools used for?

DavidFair replied:
Was from me prototyping some other packages and got left behind :+1:

DavidFair commented:
Sorry about the push spam, I noticed / fixed an issue where cppcheck would fall over since it couldn't find its rules at runtime

DavidFair commented:
Okay thanks for the reviews so far, I've:
- Added a build / push script
- Added a new deploy script (not tested locally, but it's copy paste of existing with a new image name)
- Tidied the `jenkins-node` subdir a bit

martyngigg commented:
I'll build the image and get it pushed to DockerHub.
